prototypes/uniformtree.py

```diff
@@ -434,6 +469,105 @@ def deleteRecursive(self, nodeKey):
 				continue
 			s.delete()
 
+	@exposed
+	@forceSSL
+	@forcePost
+	def move( self, skelType, key, parentNode, *args, **kwargs ):
+		"""
+		Move a node (including its contents) or a leaf to another node.
+
+		.. seealso:: :func:`canMove`
+
+		:param skelType: Defines the type of the entry that should be moved and may either be "node" or "leaf".
+		:type skelType: str
+		:param key: URL-safe key of the item to be moved.
+		:type key: str
+		:param destNode: URL-safe key of the destination node, which must be a node.
+		:type destNode: str
+
+		:returns: The rendered, edited object of the entry.
+
+		:raises: :exc:`server.errors.NotFound`, when no entry with the given *key* was found.
+		:raises: :exc:`server.errors.Unauthorized`, if the current user does not have the required permissions.
+		:raises: :exc:`server.errors.PreconditionFailed`, if the *skey* could not be verified.
+		"""
+		if "skey" in kwargs:
+			skey = kwargs["skey"]
+		else:
+			skey = ""
+		if skelType == "node":
+			skelType = TreeType.Node
+		elif skelType == "leaf" and self.leafSkelCls:
+			skelType = TreeType.Leaf
+		else:
+			raise errors.NotAcceptable()
+		skel = self.addSkel(skelType) #srcSkel
+		parentNodeSkel = self.editSkel(TreeType.Node) #destSkel
+
+
+		if not self.canMove( skelType, key, parentNode ):
+			raise errors.Unauthorized()
```

Comment:
I think we should raise here a Forbidden (403).
Unauthorized means you have submitted no or invalid credentials - But if you are logged in and have not the _move-/edit-access_ it's simply just forbidden to you.

Reply:
Valid point. This is badly handled in ViUR. To be Standard-Compliant we'd have to check if getCurrentUser() is None and raise Unauthorized(); else we should use Forbidden(). But this would also inflate the code a lot - and has no real benefit as the client already knows whenever it has a valid session or not. So for now i'd suggest we continue to use simply one (regardless of the session state) and leave this for a later release.

Reply:
Okay. I'am closing this thread and created an Issue for that: https://github.com/viur-framework/viur-core/issues/29